src/main/java/games/strategy/engine/lobby/client/ui/TimespanDialog.java

```diff
@@ -0,0 +1,107 @@
+package games.strategy.engine.lobby.client.ui;
+
+import java.awt.Component;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import javax.swing.JComboBox;
+import javax.swing.JOptionPane;
+import javax.swing.JSpinner;
+import javax.swing.SpinnerNumberModel;
+import javax.swing.border.EmptyBorder;
+import javax.swing.event.AncestorEvent;
+import javax.swing.event.AncestorListener;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import swinglib.JLabelBuilder;
+import swinglib.JPanelBuilder;
+
+/**
+ * A UI-Utility class that can be used to prompt the user for a ban or mute time.
+ */
+public class TimespanDialog {
+
+  /**
+   * The possible time units and corresponding mappings.
+   */
+  @VisibleForTesting
+  enum TimeUnit {
+    MINUTES("Minutes", i -> Instant.now().plus(i, ChronoUnit.MINUTES)),
+    HOURS("Hours", i -> Instant.now().plus(i, ChronoUnit.HOURS)),
+    DAYS("Days", i -> Instant.now().plus(i, ChronoUnit.DAYS)),
+    WEEKS("Weeks", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.WEEKS).toInstant(ZoneOffset.UTC)),
+    MONTHS("Months", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.MONTHS).toInstant(ZoneOffset.UTC)),
+    YEARS("Years", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.YEARS).toInstant(ZoneOffset.UTC)),
+    FOREVER("Forever", i -> null);
+
+    private final String name;
+    private final Function<Integer, Instant> function;
+
+    private TimeUnit(final String name, final Function<Integer, Instant> function) {
+      this.name = name;
+      this.function = function;
+    }
+
+    @Override
+    public String toString() {
+      return name;
+    }
+
+    @VisibleForTesting
+    Instant getInstant(final Integer integer) {
+      return function.apply(integer);
+    }
+  }
+
+  /**
+   * Prompts the user to enter a timespan.
+   * If the operation is not cancelled, the action Consumer is run.
+   * Not that the Date passed to the consumer can be null if the user chose forever.
+   */
+  public static void prompt(final Component parent, final String title, final String infoMessage,
+      final Consumer<Date> action) {
+    final JSpinner spinner = new JSpinner(new SpinnerNumberModel(0, 0, 99999999, 1));
+    spinner.addAncestorListener(new AncestorListener() {
```

Comment:
Unfortunately, this does not work on Linux. 😞  There's apparently a known wont-fix [Java bug](http://bugs.java.com/bugdatabase/view_bug.do?bug_id=5018574) related to this.

Reply:
Could you test it with `SwingUtilities.invokeLater(e.getComponent()::requestFocus);` ?
(Actually not sure this is correct syntax)

Reply:
Did that. :smile:  I tested with both workarounds given in the bug report, including several permutations with `invokeLater()` around various blocks of code.  No joy.

Reply:
And, yes, I tried both `requestFocus()` and `requestFocusInWindow()`. :smile:

Reply:
I really don't want to create my own custom Dialog.
Having a simple blocking method that returns the "exit" code is so handy.
If this bothers you too much, you can of course submit a PR that adds a general purpose Utility class instead of JOptionPane which offers a similar behaviour, but allows to set the default selected component

Reply:
Agreed, it's not a big deal right now.  I'll add it to my Wish List. :smile: